### PR TITLE
Enable build of System.Threading.Overlapped

### PR DESCRIPTION
**Pull request by ericstj**
/cc @weshaggard @chcosta @mellinoe

### CODE REVIEW

File: src/System.Threading.Overlapped/src/System.Threading.Overlapped.csproj [hunk 1]
```diff
@@ -41,8 +37,14 @@
       <Link>Common\NotImplemented.cs</Link>
     </Compile>
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetGroup)'=='netstandard'">
+  <ItemGroup Condition="'$(TargetGroup)'=='netstandard' AND '$(TargetsWindows)' == 'true'">
     <TargetingPackReference Include="System.Private.CoreLib" />
+    <Reference Include="System.Private.CoreLib" />
```
Comment (ericstj) [inline src/System.Threading.Overlapped/src/System.Threading.Overlapped.csproj]: I noticed that all of the partial facades require this redundant ref to trigger https://github.com/dotnet/buildtools/blob/7b3991e4bba0b86945a06470b313f9183e744e25/src/Microsoft.DotNet.Build.Tasks/PackageFiles/packageresolve.targets#L154.

@weshaggard mentioned we should just turn that on by default.  I've filed https://github.com/dotnet/corefx/issues/14669 to clean these up.

**Comment by ericstj** [comment]
Fixes #14573

**Comment by mellinoe** [comment]
LGTM

File: src/src.builds [hunk 2]
```diff
@@ -3,7 +3,6 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <ItemGroup>
     <ProjectExclusions Condition="'$(OS)' != 'Windows_NT'" Include="$(MSBuildThisFileDirectory)**\src\System.Memory.csproj" />
```
Comment (weshaggard) [inline src/src.builds]: My expectation is that we don't need this exclusion either.

**Comment by weshaggard** [comment]
LGTM

**Pull request merged**